### PR TITLE
docs: added cw721-poap contract README.md

### PR DESCRIPTION
**Pull request by dadamu**
This PR added README.md to cw721-poap.

### CODE REVIEW

File: contracts/cw721-poap/README.md [hunk 1]
```diff
@@ -0,0 +1,529 @@
...
+}
+```
+
+### SendNft
+Allows to move a token to another contract then trigger an action. This message has the following parameters:
+* `contract`: Contract address where the token transfer to;
+* `token_id`: Id of the token which would be transferred;
+* `msg`: Base64 encoded message to trigger on the receiver contract.
+
+An example of the message to send nft having a trigger message:
```
Comment (manu0466) [inline contracts/cw721-poap/README.md:45]:
```suggestion
Here an example message to send nft having a trigger message:
```

File: contracts/cw721-poap/README.md [hunk 2]
```diff
@@ -0,0 +1,529 @@
...
+        "contract": "desmos1......",
+        "token_id": "1",
+        "msg": "eyJleGVjdXRlX2V4YW1wbGUiOnt9fQ=="
+    }
+}
+```
+**Note** the msg is base64-encoded of `{"execute_example":{}}`, which is the sample execution message on the target contract. 
+
+### Approve
+Allows to give a user an access to transfer/send the token from the owner's account. This message has the following parameters:
```
Comment (manu0466) [inline contracts/cw721-poap/README.md:58]:
```suggestion
Allows a user to transfer/send the token from the owner's account. This message has the following parameters:
```

File: contracts/cw721-poap/README.md [hunk 3]
```diff
@@ -0,0 +1,529 @@
...
+* `spender`: Address who would be assigned as an access of the token;
+* `token_id`: Id of the target token;
+* `expires`: Expiration time/height of this allowance, if it is set as `null` then it has no time/height limit.
+
+An example of the message to approve a user to have the sending/transferring access to a token with an expiration height:
```
Comment (manu0466) [inline contracts/cw721-poap/README.md:63]:
```suggestion
An example message to approve a user to have the sending/transferring access to a token with an expiration height:
```

File: contracts/cw721-poap/README.md [hunk 4]
```diff
@@ -0,0 +1,529 @@
...
+        "spender": "desmos1......",
+        "token_id": "1",
+        "expires": {
+            "at_height": 1000
+        }
+    }
+}
+```
+
+An example of the message to approve a token to an operator with an expiration time:
```
Comment (manu0466) [inline contracts/cw721-poap/README.md:76]:
```suggestion
Here an example message to approve a token to an operator with an expiration time:
```

File: contracts/cw721-poap/README.md [hunk 5]
```diff
@@ -0,0 +1,529 @@
...
+        "spender": "desmos1......",
+        "token_id": "1",
+        "expires": {
+            "at_time": "2022-01-01T00:00:00Z"
+        }
+    }
+}
+```
+
+An example of the message to approve a token to an operator without any expiration:
```
Comment (manu0466) [inline contracts/cw721-poap/README.md:89]:
```suggestion
Here an example message to approve a token to an operator without any expiration:
```

File: contracts/cw721-poap/README.md [hunk 6]
```diff
@@ -0,0 +1,529 @@
...
+    }
+}
+```
+
+### Revoke
+Allows to remove a previously granted approval. This message has the following parameters:
+* `spender`: Address who would be revoked the permission of the given token;
+* `token_id`: Id of the target token.
+
+An example of the message to revoke an operator to a token:
```
Comment (manu0466) [inline contracts/cw721-poap/README.md:105]:
```suggestion
Here an example message to revoke an operator to a token:
```

File: contracts/cw721-poap/README.md [hunk 7]
```diff
@@ -0,0 +1,529 @@
...
+    }
+}
+```
+
+### ApproveAll
+Allows to give all the tokens transferring/sendind tokens approval to an operator from the owner's account. This message has the following parameters:
+* `operator`: Address who is assigned to have all the tokens approvals in the owner's account;
+* `expires`: Expiration time/height of this allowance, if it is set as `null` then it has no time/height limit.
+
+An example of the message to approve an operator all the tokens with an expiration height:
```
Comment (manu0466) [inline contracts/cw721-poap/README.md:120]:
```suggestion
Here an example message to approve an operator all the tokens with an expiration height:
```

File: contracts/cw721-poap/README.md [hunk 8]
```diff
@@ -0,0 +1,529 @@
...
+    "approve_all": {
+        "spender": "desmos1......",
+        "expires": {
+            "at_height": 1000
+        }
+    }
+}
+```
+
+An example of the message to approve an operator all the tokens with an expiration time:
```
Comment (manu0466) [inline contracts/cw721-poap/README.md:132]:
```suggestion
Here an example message to approve an operator all the tokens with an expiration time:
```

File: contracts/cw721-poap/README.md [hunk 9]
```diff
@@ -0,0 +1,529 @@
...
+    "approve_all": {
+        "spender": "desmos1......",
+        "expires": {
+            "at_time": "2022-01-01T00:00:00Z"
+        }
+    }
+}
+```
+
+An example of the message to approve an operator all the tokens without any expiration:
```
Comment (manu0466) [inline contracts/cw721-poap/README.md:144]:
```suggestion
Here an example message to approve an operator all the tokens without any expiration:
```

File: contracts/cw721-poap/README.md [hunk 10]
```diff
@@ -0,0 +1,529 @@
...
+        "expires": null
+    }
+}
+```
+
+### RevokeAll
+Allows to remove a previously granted approval all permission. This message has the following parameters:
+* `operator`: Address who would be revoked operator permissions of all the tokens from the owner's account.
+
+An example of the message to revoke operator permissions to all the tokens from the owner's account:
```
Comment (manu0466) [inline contracts/cw721-poap/README.md:158]:
```suggestion
Here an example message to revoke operator permissions to all the tokens from the owner's account:
```

File: contracts/cw721-poap/README.md [hunk 11]
```diff
@@ -0,0 +1,529 @@
...
+```
+
+### Mint
+Allows the minter to mint a new NFT to a user. This message has the following parameters:
+* `token_id`: unique id of the NFT;
+* `owner`: the owner of the newly minted NFT;
+* `token_uri`: universal resource identifier for this NFT;
+* `extension`: the `POAP metadata` which includes claimer of this NFT.
+
+An example of the message to mint new NFT:
```
Comment (manu0466) [inline contracts/cw721-poap/README.md:174]:
```suggestion
Here an example message to mint a new NFT:
```

File: contracts/cw721-poap/README.md [hunk 12]
```diff
@@ -0,0 +1,529 @@
...
+        }
+    }
+}
+```
+
+### Burn
+Allows to burn an NFT the sender has access to. This message has the following parameters:
+* `token_id`: Id of the token that would be burned.
+
+An example of the message to burn an NFT:
```
Comment (manu0466) [inline contracts/cw721-poap/README.md:192]:
```suggestion
Here an example message to burn an NFT:
```

File: contracts/cw721-poap/README.md [hunk 13]
```diff
@@ -0,0 +1,529 @@
...
+```
+
+## Query Messages
+
+### OwnerOf
+Returns the owner of the given token, error if token does not exist. This message has the following parameters:
+* `token_id`: Id of the target token;
+* `include_expired`: Trigger to filter out expired approvals, unset or false will exclude expired approvals.
+
+Here an example of the message to query the owner of the given token:
```
Comment (manu0466) [inline contracts/cw721-poap/README.md:208]:
```suggestion
Here an example message to query the owner of the given token:
```

File: contracts/cw721-poap/README.md [hunk 14]
```diff
@@ -0,0 +1,529 @@
...
+}
+```
+
+### Approval
+Returns an access owned by the given spender to the given token. This message has the following parameters:
+* `token_id`: Id of the target token;
+* `spender`: Address who has the sending/transferring access to the given token;
+* `include_expired`: Trigger to filter out expired approvals, unset or false will exclude expired approvals.
+
+Here an example of the message to query the approval of the given token by a spender:
```
Comment (manu0466) [inline contracts/cw721-poap/README.md:251]:
```suggestion
Here an example message to query the approval of the given token by a spender:
```

File: contracts/cw721-poap/README.md [hunk 15]
```diff
@@ -0,0 +1,529 @@
...
+    } 
+}
+```
+
+### Approvals
+Returns approvals that a token has. This message has the following parameters:
+* `token_id`: Id of the target token;
+* `include_expired`: Trigger to filter out expired approvals, unset or false will exclude expired approvals.
+
+Here an example of the message to query the approvals of the given token:
```
Comment (manu0466) [inline contracts/cw721-poap/README.md:279]:
```suggestion
Here an example message to query the approvals of the given token:
```

File: contracts/cw721-poap/README.md [hunk 16]
```diff
@@ -0,0 +1,529 @@
...
+```
+
+### AllOperators
+Lists all operators that can access all of the owner's tokens. This message has the following parameters:
+* `owner`: Address of the owner to be queried.
+* `include_expired`: Trigger to filter out expired approvals, unset or false will exclude expired approvals;
+* `start_after`: Position in address where tokens start after;
+* `limit`: Limitation to list the number of operators, if unset would be 10 and the maximum is 100.
+
+Here an example of the message to query the operators of the given owner:
```
Comment (manu0466) [inline contracts/cw721-poap/README.md:322]:
```suggestion
Here an example message to query the operators of the given owner:
```

File: contracts/cw721-poap/README.md [hunk 17]
```diff
@@ -0,0 +1,529 @@
...
+            }
+        },
+    ]
+}
+```
+
+### NumTokens
+Returns total number of tokens issued.
+
+Here an example of the message to query total number of tokens:
```
Comment (manu0466) [inline contracts/cw721-poap/README.md:363]:
```suggestion
Here an example message to query total number of tokens:
```

File: contracts/cw721-poap/README.md [hunk 18]
```diff
@@ -0,0 +1,529 @@
...
+```json
+{
+    "count": 1000
+}
+```
+
+### ContractInfo
+Returns top-level metadata about the contract.
+
+Here an example of the message to query the contract info of the contract:
```
Comment (manu0466) [inline contracts/cw721-poap/README.md:380]:
```suggestion
Here an example message to query the contract info of the contract:
```

File: contracts/cw721-poap/README.md [hunk 19]
```diff
@@ -0,0 +1,529 @@
...
+    "name": "test_name",
+    "symbol": "test_symbol"
+}
+```
+
+### NftInfo
+Returns metadata about one particular token. This message has the following parameters:
+* `token_id`: Id of the target token.
+
+Here an example of the message to query the info of the given token:
```
Comment (manu0466) [inline contracts/cw721-poap/README.md:399]:
```suggestion
Here an example message to query the info of the given token:
```

File: contracts/cw721-poap/README.md [hunk 20]
```diff
@@ -0,0 +1,529 @@
...
+{
+    "token_uri": "ipfs://token.erc721.metadata",
+    "extension": {
+        "claimer": "desmos1......"
+    }
+}
+```
+
+### AllNftInfo
+Returns the result of both `NftInfo` and `OwnerOf` as one query as an optimization. This message has the following parameters:
```
Comment (manu0466) [inline contracts/cw721-poap/README.md:419]:
```suggestion
Returns the result of both `NftInfo` and `OwnerOf`. This message has the following parameters:
```

File: contracts/cw721-poap/README.md [hunk 21]
```diff
@@ -0,0 +1,529 @@
...
+* `token_id`: Id of the target token.
+* `include_expired`: Trigger to filter out expired approvals, unset or false will exclude expired approvals.
+
+Here an example of the message to query all the info of the given token:
```
Comment (manu0466) [inline contracts/cw721-poap/README.md:423]:
```suggestion
Here an example message to query all the info of the given token:
```

File: contracts/cw721-poap/README.md [hunk 22]
```diff
@@ -0,0 +1,529 @@
...
+}
+```
+
+### Tokens
+Returns all tokens owned by the given address. This message has the following parameters:
+* `owner`: Target address owned tokens to be queried;
+* `start_after`: Position in token id where tokens start after;
+* `limit`: Limitation to list the number of tokens, if unset would be 10 and the maximum is 100.
+
+Here an example of the message to query all the tokens owned by the given address:
```
Comment (manu0466) [inline contracts/cw721-poap/README.md:474]:
```suggestion
Here an example message to query all the tokens owned by the given address:
```

File: contracts/cw721-poap/README.md [hunk 23]
```diff
@@ -0,0 +1,529 @@
...
+    "tokens": ["2", "3", "4"]
+}
+```
+
+### AllTokens
+Lists all token_ids in the contract. This message has the following parameters:
+* `start_after`: Position in token id where tokens start after;
+* `limit`: Limitation to list the number of tokens, if unset would be 10 and the maximum is 100.
+
+Here an example of the message to query all the tokens in the contract:
```
Comment (manu0466) [inline contracts/cw721-poap/README.md:497]:
```suggestion
Here an example message to query all the tokens in the contract:
```

File: contracts/cw721-poap/README.md [hunk 24]
```diff
@@ -0,0 +1,529 @@
...
+```json
+{
+    "tokens": ["2", "3", "4"]
+}
+```
+
+### Minter
+Returns the minter who is the one having access to mint NFT.
+
+Here an example of the message to query the minter of the contract:
```
Comment (manu0466) [inline contracts/cw721-poap/README.md:517]:
```suggestion
Here an example message to query the minter of the contract:
```

File: contracts/cw721-poap/README.md [hunk 25]
```diff
@@ -0,0 +1,529 @@
...
+```
+
+## Query Messages
+
+### OwnerOf
+Returns the owner of the given token, error if token does not exist. This message has the following parameters:
+* `token_id`: Id of the target token;
+* `include_expired`: Trigger to filter out expired approvals, unset or false will exclude expired approvals.
+
+Here Here an example meesage to query the owner of the given token:
```
Comment (manu0466) [inline contracts/cw721-poap/README.md:208]:
```suggestion
Here an example meesage to query the owner of the given token:
```

File: contracts/cw721-poap/README.md [hunk 26]
```diff
@@ -0,0 +1,529 @@
...
+}
+```
+
+### Approval
+Returns an access owned by the given spender to the given token. This message has the following parameters:
+* `token_id`: Id of the target token;
+* `spender`: Address who has the sending/transferring access to the given token;
+* `include_expired`: Trigger to filter out expired approvals, unset or false will exclude expired approvals.
+
+Here Here an example meesage to query the approval of the given token by a spender:
```
Comment (manu0466) [inline contracts/cw721-poap/README.md:251]:
```suggestion
Here an example meesage to query the approval of the given token by a spender:
```

File: contracts/cw721-poap/README.md [hunk 27]
```diff
@@ -0,0 +1,529 @@
...
+    } 
+}
+```
+
+### Approvals
+Returns approvals that a token has. This message has the following parameters:
+* `token_id`: Id of the target token;
+* `include_expired`: Trigger to filter out expired approvals, unset or false will exclude expired approvals.
+
+Here Here an example meesage to query the approvals of the given token:
```
Comment (manu0466) [inline contracts/cw721-poap/README.md:279]:
```suggestion
Here an example meesage to query the approvals of the given token:
```

File: contracts/cw721-poap/README.md [hunk 28]
```diff
@@ -0,0 +1,529 @@
...
+```
+
+### AllOperators
+Lists all operators that can access all of the owner's tokens. This message has the following parameters:
+* `owner`: Address of the owner to be queried.
+* `include_expired`: Trigger to filter out expired approvals, unset or false will exclude expired approvals;
+* `start_after`: Position in address where tokens start after;
+* `limit`: Limitation to list the number of operators, if unset would be 10 and the maximum is 100.
+
+Here Here an example meesage to query the operators of the given owner:
```
Comment (manu0466) [inline contracts/cw721-poap/README.md:322]:
```suggestion
Here an example meesage to query the operators of the given owner:
```

File: contracts/cw721-poap/README.md (hunk 25 above)
Comment (dadamu) [reply contracts/cw721-poap/README.md:208]: Sorry by the mistake of replace all. 🙏

File: contracts/cw721-poap/README.md [hunk 29]
```diff
@@ -0,0 +1,529 @@
...
+To easily interact with the contract you can use the `cw721-poap` script available [here](https://github.com/desmos-labs/contract-utils/tree/main/utils), 
+otherwise you can take a look at the supported messages in the following sections.
+
+## Instantiate Message
+Allows to initialize the contract. This message has the following parameters:
+* `name`: Name of the NFT contract;
+* `symbol`: Symbol of the NFT contract;
+* `minter`: Address who is the only one to be able to create new NFTs.
+
+Here an example of instantiate message:
```
Comment (dadamu) [inline contracts/cw721-poap/README.md:13]:
```suggestion
Here an example message to instantiate a contract:
```

**Review by manu0466 — APPROVED** [review]

**Review by bragaz — APPROVED** [review]

**Pull request merged**